src/provider.rs

```diff
@@ -226,33 +236,42 @@ pub fn get_provider_by_domain(domain: &str) -> Option<&'static Provider> {
 ///
 /// For security reasons, only Gmail can be configured this way.
 pub async fn get_provider_by_mx(context: &Context, domain: &str) -> Option<&'static Provider> {
-    if let Ok(resolver) = get_resolver() {
-        let mut fqdn: String = domain.to_string();
-        if !fqdn.ends_with('.') {
-            fqdn.push('.');
+    let Ok(resolver) = get_resolver() else {
+        warn!(context, "Cannot get a resolver to check MX records.");
+        return None;
+    };
+
+    let mut fqdn: String = domain.to_string();
+    if !fqdn.ends_with('.') {
+        fqdn.push('.');
+    }
+
+    let Ok(mx_domains) = resolver.mx_lookup(fqdn).await else {
+        warn!(context, "Cannot resolve MX records for {domain:?}.");
+        return None;
+    };
+
+    for (provider_domain_pattern, provider) in PROVIDER_DATA.as_slice() {
```

Comment:
Why `as_slice()` is needed here while it's not needed in `get_provider_by_domain()`?